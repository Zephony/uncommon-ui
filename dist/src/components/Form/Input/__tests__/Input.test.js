"use strict";

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@testing-library/react");

var _styledComponents = require("styled-components");

var _ = _interopRequireDefault(require("../"));

var _theme = _interopRequireDefault(require("../../../../utils/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe("Input", function () {
  var labelText = "This is a label";
  var props = {
    type: "text",
    label: labelText
  };
  it("should render an input component", function () {
    var _render = (0, _react2.render)( /*#__PURE__*/_react.default.createElement(_styledComponents.ThemeProvider, {
      theme: _theme.default
    }, /*#__PURE__*/_react.default.createElement(_.default, props))),
        container = _render.container;

    expect(container.firstChild).toMatchSnapshot();
  });
  it("should contain a label", function () {
    var _render2 = (0, _react2.render)( /*#__PURE__*/_react.default.createElement(_styledComponents.ThemeProvider, {
      theme: _theme.default
    }, /*#__PURE__*/_react.default.createElement(_.default, props))),
        getByText = _render2.getByText;

    expect(getByText(labelText)).toBeTruthy();
  });
});

//# sourceMappingURL=Input.test.js.map