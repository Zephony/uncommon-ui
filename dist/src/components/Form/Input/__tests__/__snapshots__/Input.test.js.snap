// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render an input component 1`] = `
<div>
  <label
    class="sc-AxirZ iRQJOi"
  >
    This is a label
  </label>
  <div
    class="sc-AxjAm bCruhs"
  >
    <input
      class="sc-AxiKw jpCyRs"
      type="text"
    />
  </div>
</div>
`;
